calcom/cal.com
# perf: pre-load booking data for all users

## What does this PR do?

Instead of loading booking data per user when determining availability, we now preload it for all users. The query currently loading booking data in the loop for all users is actually the slowest query in our system according to database load data.

## Type of change




## How should this be tested?



- Are there environment variables that should be set? **No**
- What are the minimal test data to have? **A round robin event with 20 users minimum. Ideally more.**
- What is expected (happy path) to have (input and output)? **Booking calendar availability calculations don't change at all. They are just incredibly faster for large round robin events.**

## Mandatory Tasks
